CustomBatchWatchdog.cs

```diff
@@ -207,13 +218,36 @@ private void RunForever()
             }
         }
 
+
+
         protected override void OnStart(string[] args)
         {
             //System.Diagnostics.Debugger.Launch();
             PrintInfo("Custom batch watchdog has been started.");
+
+            if (args.Length > 0)
+            {
+                OverrideSettings(args);
+            }
+
             LoadConfigFromFile();
             ThreadPool.QueueUserWorkItem(o => { RunForever(); });
         }
+        private void OverrideSettings(string[] args)
+        {
+            if (args.Length > 0)
+            {
+                configFileName = args[0];
+                if (configFileName.IndexOf('/') > -1)
+                    configFileName = configFileName.Replace("/", string.Empty);
```

Comment:
Use the [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) for braces

```csharp
if (configFileName.IndexOf('/') > -1)
{
    configFileName = configFileName.Replace("/", string.Empty);
}
```

We usually use `-` (short version) and `--` (long version) for CLI options, see `star.exe`:

```
C:\Github\Starcounter\cbwatchdog [Iqbal ≡]> star.exe
Usage: star [options] executable [parameters]

Options:
  -h, --help            Shows help about star.exe.
  -hx, --helpextended   Shows extended help about star.exe.
  -v, --version         Prints the version of Starcounter.
  -i, --info            Prints information about the Starcounter installation.
  -p=,--serverport=port Port to the admin server.
  --server=name         Specifies the name of the server. If no port is
                        specified, star.exe use the known port of server.
  --serverhost=host     Specifies the identity of the server host. If no
                        host is specified, 'localhost' is used.
  -d=,--database=name   The database to use. 'Default' is used if not given.
                        Example "star d=foo bar.exe"
  --stop                Stops the given application.
  --norestart           Prevent the application from being restarted if running.
  --name=name           Gives the application the specified name.
  -t, --transact        Runs the entrypoint in the scope of a write transaction
```

or `stardump.exe unload -?`

```
C:\Github\Starcounter\cbwatchdog [Iqbal ≡]> stardump unload -?


Usage:  unload [options]

Options:
  -db | --database <DatabaseName>               Name of starcounter database to dump. Default value is "default"
  -f | --file <FileName>                        Output dump filename
  -b | --buffersize <BufferSize>                Set insert rows buffer size to dump database.
  -scp | --skipcolumnprefixes <ColumnPrefixes>  Column prefixes to skip, space and/or comma separated. Example: -scp="a b,c". Default value is "__"
  -stp | --skiptableprefixes <TablePrefixes>    Table prefixes to skip, space and/or comma separated. Example: -stp="a b,c"
  -st | --skiptables <TableNames>               Table names to skip, space and/or comma separated. Example: -stp="a b,c"
  -ut | --unloadtables <TableNames>             Table names to unlaod, space and/or comma separated. Example: -stp="a b,c"
  -V | --verbose <Verbose>                      Verbose mode, higher number prints more. Default number is 2 [Error:0, Warning:1, Notice:2, Info:3, Debug:4]
  -? | -h | --help                              Show help information
```

Reply:
The net start command doesn't accept both - and --. It only accepts /